armi/nuclearDataIO/cccc/dif3d.py

```diff
@@ -0,0 +1,263 @@
+    "PSINRM",
+    "POWIN",
+    "SIGBAR",
+    "EFFKQ",
+    "EPSWP",
+] + [f"DUM{e}" for e in range(1, 21)]
+
+
+class Dif3dData(cccc.DataContainer):
+    def __init__(self):
+        cccc.DataContainer.__init__(self)
+
+        self.twoD = {e: None for e in FILE_SPEC_2D_PARAMS}
+        self.threeD = {e: None for e in FILE_SPEC_3D_PARAMS}
+        self.fourD = None
+        self.fiveD = None
+
+
+class Dif3dStream(cccc.StreamWithDataContainer):
+    @staticmethod
+    def _getDataContainer() -> Dif3dData:
+        return Dif3dData()
+
+    def _rwFileID(self):
+        """
+        Record for file identification information.
+
+        The parameters are stored as a dictionary under the attribute `metadata`.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+        with self.createRecord() as record:
+            for param in ["HNAME", "HUSE1", "HUSE2"]:
+                self._metadata[param] = record.rwString(self._metadata[param], 8)
+            self._metadata["VERSION"] = record.rwInt(self._metadata["VERSION"])
+
+    def _rw1DRecord(self):
+        """
+        Record for problem title, storage, and dump specifications.
+
+        The parameters are stored as a dictionary under the attribute `metadata`.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+        with self.createRecord() as record:
+            for i in range(11):
+                param = f"TITLE{i}"
+                self._metadata[param] = record.rwString(self._metadata[param], 8)
+            self._metadata["MAXSIZ"] = record.rwInt(self._metadata["MAXSIZ"])
+            self._metadata["MAXBLK"] = record.rwInt(self._metadata["MAXBLK"])
+            self._metadata["IPRINT"] = record.rwInt(self._metadata["IPRINT"])
+
+    def _rw2DRecord(self):
+        """
+        Record for DIF3D integer control parameters.
+
+        The parameters are stored as a dictionary under the attribute `twoD`.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+        with self.createRecord() as record:
+            for param in FILE_SPEC_2D_PARAMS:
+                self._data.twoD[param] = record.rwInt(self._data.twoD[param])
+
+    def _rw3DRecord(self):
+        """
+        Record for convergence criteria and other sundry floating point data (such as
+        k-effective).
+
+        The parameters are stored as a dictionary under the attribute `threeD`.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+        with self.createRecord() as record:
+            for param in FILE_SPEC_3D_PARAMS:
+                self._data.threeD[param] = record.rwDouble(self._data.threeD[param])
+
+    def _rw4DRecord(self):
+        """
+        Record for the optimum overrelaxation factors. This record is only present if `NUMORP`
```

Comment:
In some additional learnings I just received, my understanding is that this record is only written out when using DIF3D-FD (finite difference). We could add that to the docstring.